### PR TITLE
Start emulator on `flow dev` with flag

### LINKED ISSUE #1392: `flow dev` should start `flow emulator`
Currently, to use `flow dev` you have to start the emulator and then also start `flow dev`. Considering this is supposed to be a way to simplify development, it would be nice to reduce complexity. The only potential issue I can see is with state. Although, I don't think that people using `flow dev` are generally considered with starting from a new state each time given the way accounts are created. @bluesign any thoughts?

### PR DESCRIPTION
**Pull request by chasefleming**
Closes #1392
______

For contributor use:

- [x] Targeted PR against `master` branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [ ] Code follows the [standards mentioned here](https://github.com/onflow/flow-cli/blob/master/CONTRIBUTING.md#styleguides)
- [ ] Updated relevant documentation
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

**Review by ianthpun — APPROVED** [review]

File: internal/super/dev.go [hunk 1]
```diff
@@ -61,22 +68,60 @@ func dev(
...
+			logger.Error("Error connecting to emulator. Make sure you started an emulator using 'flow emulator' command.")
+			logger.Info(fmt.Sprintf("%s This tool requires emulator to function. Emulator needs to be run inside the project root folder where the configuration file ('flow.json') exists.\n\n", output.TryEmoji()))
+			return nil, nil
+		}
+	} else {
+		err = flow.Ping()
+		if err != nil {
+			logger.Error("Error connecting to emulator. Make sure you started an emulator using 'flow emulator' command.")
+			logger.Info(fmt.Sprintf("%s This tool requires emulator to function. Emulator needs to be run inside the project root folder where the configuration file ('flow.json') exists.\n\n", output.TryEmoji()))
+			return nil, nil
```
Comment (jribbink) [inline internal/super/dev.go:117]:
```suggestion
			return nil, err
```

File: internal/super/dev.go [hunk 2]
```diff
@@ -61,22 +68,60 @@ func dev(
...
+			<-c
+			os.Exit(1)
+		}()
+
+		ctx := context.Background()
+		err = flow.WaitServer(ctx)
+		if err != nil {
+			logger.Error("Error connecting to emulator. Make sure you started an emulator using 'flow emulator' command.")
+			logger.Info(fmt.Sprintf("%s This tool requires emulator to function. Emulator needs to be run inside the project root folder where the configuration file ('flow.json') exists.\n\n", output.TryEmoji()))
+			return nil, nil
```
Comment (jribbink) [inline internal/super/dev.go:110]:
```suggestion
			return nil, err
```

File: internal/super/dev.go [hunk 3]
```diff
@@ -61,22 +68,60 @@ func dev(
...
+		emulatorServer := server.NewEmulatorServer(&log, &server.Config{
+			ServicePublicKey:   (*privateKey).PublicKey(),
+			ServicePrivateKey:  *privateKey,
+			ServiceKeySigAlgo:  serviceAccount.Key.SigAlgo(),
+			ServiceKeyHashAlgo: serviceAccount.Key.HashAlgo(),
+		})
+
+		emuErr := emulatorServer.Listen()
+		if emuErr != nil {
+			panic(fmt.Sprintf("Failed to prepare the emulator for connections: %s", emuErr))
```
Comment (jribbink) [inline internal/super/dev.go:94]:
```suggestion
			return nil, fmt.Errorf("Failed to prepare the emulator for connections: %s", emuErr)
```

**Review by jribbink — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -44,4 +44,6 @@ jobs:
       - uses: golangci/golangci-lint-action@v3.7.0
         with:
           version: v1.52.2
+          only-new-issues: true
```
Comment (ianthpun) [inline .github/workflows/ci.yml:47]: this'll fix the log dump on ci/cd lint failure

**Pull request merged**